rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java

```diff
@@ -195,7 +195,7 @@ boolean matchesTargetType(@Nullable JavaType.FullyQualified type) {
 
         if (targetTypePattern.matcher(type.getFullyQualifiedName()).matches()) {
             return true;
-        } else if (!type.getFullyQualifiedName().equals("java.lang.Object") && matchesTargetType(type.getSupertype() == null ? JavaType.Class.build("java.lang.Object") : type.getSupertype())) {
+        } else if (!type.getFullyQualifiedName().equals("java.lang.Object") && matchesTargetType(JavaType.Class.build("java.lang.Object"))) {
```

Comment:
We want the "...otherwise use the super"-logic, but we should only check the supertype if we specify `matchOverrides`.

On L198 here, the current code of:

```java
matchesTargetType(type.getSupertype() == null ? JavaType.Class.build("java.lang.Object") : type.getSupertype())
```

will recursively check if any of the (current type being checked)'s supertypes match. But we only want this to happen if `matchesOverrides` is `true`. The way this code is currently, this will _always_ check if any of the supertypes match.

If we look at the lines below, we perform this same logic within the `matchOverrides` block: https://github.com/openrewrite/rewrite/blob/fe05e191bd7259398fe0a7ff1efe03bbb71c4d60/rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java#L202-L205